### PR TITLE
Vault decrypt on PRs

### PR DESCRIPTION
**Pull request by nitay**
Decrypt Ansible Vault files in pull requests (in addition to blobs which already work). This works by looking for diffs like:

```
$ANSIBLE_VAULT;1.1;AES256
- 6163363461303763336
- 6236663635633134663
+ 3465636531343132386
+ 3865643830663865643
```

Specifically, the ansible vault header, followed by the entire old file, followed by entire new one. Once found, it decrypts both files, does a diff (using JS diff library I pulled in), and then replaces the text. A lot of the code here is applying various classes/ids to make it "look like GitHub".

Note I wrote this all using ES6.

Caveats:
1) This always outputs in "Unified" mode. If you view a PR in "Split" mode it will look weird.
2) Commenting not supported. Adding the button to enable comments is pretty easy I figured out how to do that, but the hard part is once someone writes a comment, having to pick it up the whole thread and displaying old comments after decrypting. So for now I just disabled comments.

Once you guys approve I'll also submit it upstream.

### CODE REVIEW

File: common.js [hunk 1]
```diff
@@ -49,3 +49,47 @@ function if_auto_decrypt(cb) {
...
+// [B](f: (A) ⇒ [B]): [B]  ; Although the types in the arrays aren't strict (:
+Array.prototype.flatMap = function(lambda) {
+    return Array.prototype.concat.apply([], this.map(lambda));
+};
+
+function sanitize(node) {
+    return node.text().trim().replace(/\s/g, '');
+}
+
+function handle_response(textToDecrypt, response, prompt_on_fail, callback) {
```
Comment (ebaronov) [inline common.js]: In general we don't pass callbacks anymore in JS. We use Promises to handle async computation. However, this is not really production code so I think it's fine.

File: github-diff.js [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+
+  // We're assuming a diff on vault file will have following format:
+  //
+  // $ANSIBLE_VAULT;1.1;AES256          (ansible vault header)
+  // - DELETIONS                        (all the deletions)
+  // + ADDITIONS                        (all the additions)
+  //
+  static detect() {
+    const files = $('#files .file.js-file');
+    console.log('XX detect');
```
Comment (ebaronov) [inline github-diff.js]: remove?

File: common.js (hunk 1 above)
Comment (nitay) [inline common.js]: gotcha, yeah the whole rest of this code was written without them don't want to hack that all up yet, but good to know thx

File: github-diff.js (hunk 2 above)
Comment (nitay) [inline github-diff.js]: oops will do

File: common.js [hunk 3]
```diff
@@ -49,3 +49,47 @@ function if_auto_decrypt(cb) {
...
+      if (prompt_on_fail) {
+        display_error(response.error);
+      }
+  }
+}
+
+function prompt_and_decrypt(textToDecrypt, callback) {
+  prompt_password(function(pw) {
+    do_decrypt(textToDecrypt, pw, true, callback);
+  });
```
Comment (harryaiq) [inline common.js]: if you wanted to use more es6 most of these functions could be rewritten as
```
pw => do_decrypt(textToDecrypt, pw, true, callback)
```

File: github-blob.js [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+  }
+}
+
+class Blob {
+  constructor(blobLines) {
+    this.blobLines = blobLines;
+  }
+
+  static compute(lines) {
+    const blobLines = lines.split('\n').map((line, idx) => { return new BlobLine(idx, line) });
```
Comment (harryaiq) [inline github-blob.js]:
```
const blobLines = lines.split('\n').map((line, idx) => new BlobLine(idx, line));
```
more es6 stuff -- if only one line of code, no need for `{` `}` or the return

**Comment by harryaiq** [comment]
this looks good. really hard to write organized/concise code when building it on top of a current site. the only suggestion would be to move most of the github assumed classes/ids to a constant file so when things break in the future it might be easier to change.

**Comment by antkim003** [comment]
yeah nitay, this lgtm

**Pull request merged**